### DISCUSSION TITLE
How to define ClientOnly globally for components from UI library (npm package)?

### DISCUSSION THREAD

**Discussion by hrynevychroman** [nuxt/nuxt#28536] [Questions]
For example, I have something like
```ts
import { Dropdown } from 'ui-library';
```

My `ui-library` is based on Vue library mode using Bootstrap, which is not SSR-friendly. I want to tell Nuxt that every `Dropdown` from this `ui-library` (npm package) needs to be `<ClientOnly>`. Don't want to create a wrapper component for this or wrap every `Dropdown` in all related templates 😅

I don't see any configuration, but this may be great for such a use case.
```ts
// nuxt.config.ts
export default defineNuxtConfig({
	components: [
		{
			name: 'Dropdown',
			filePath: 'ui-library',
			mode: 'client'
		}
	]
});
```

**Comment by xjccc** [comment]
You can use `.client` suffix. Like `MyComponent.client.vue` for create a `client only component`

**Reply by hrynevychroman** [reply]
This is for locally defined components, but I am asking for components imported from the npm package, third party library ☺️

**Reply by xjccc** [reply]
Like this to register a local module.
[npm packages](https://nuxt.com/docs/guide/directory-structure/components#npm-packages)

**Reply by hrynevychroman** [reply]
yes, but how I specify that this component is client-only?

**Reply by xjccc** [reply]
Like Your code
```
{
   mode: 'client'
}
```

**Reply by hrynevychroman** [reply]
I can't use setup() inside `defineNuxtConfig` 🤔

**Reply by xjccc** [reply]
In your `modules/` to register a module. Like this [npm packages](https://nuxt.com/docs/guide/directory-structure/components#npm-packages)
```
import { addComponent, defineNuxtModule } from '@nuxt/kit'

export default defineNuxtModule({
  setup() {
    // import { MyComponent as MyAutoImportedComponent } from 'my-npm-package'
    addComponent({
      name: 'Dropdown',
      filePath: 'ui-library',
      mode: 'client'
    })
  }
})

```

**Reply by hrynevychroman** [reply]
Yes, if I define this component as a module, it will work fine; how could I import all other components from this package, too? Because it is slightly misunderstood if I `import { Dropdown }` it will not work, but if use it without import - works great 😁 

Also, this will not cause some issue with bundler size, yes?

**Reply by xjccc** [reply]
I don't know this.. I think something like other ui lib's imports [vant-nuxt](https://github.com/vant-ui/vant-nuxt/blob/main/src/core/components.ts)

**Reply by hrynevychroman** [reply]
Big thanks, will take a look and get with some response 🙌
